Product overview changes.

alexander-huerta opened the pull request:
Also downloaded 3 loaders into the dev dependencies for css. Reconfigured webpack to handle css files and images.

RehReis commented on client/src/components/ProductDetails/ImageGallery.jsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import css from './prodStyles.css';
```
So I can understand, why does this need to be imported here (and in the other files)? Is this needed everywhere?

alexander-huerta replied:
the css file needs to be imported into each component that intends to style. I just deleted the import statements from the files not using css. 

https://github.com/webpack-contrib/css-loader

It can also be more concisely imported using this syntax "import './prodStyles.css';
" -- I just changed the various import statements.

RehReis commented on client/src/components/ProductDetails/ImageThumbnail.jsx:
```diff
@@ -0,0 +1,16 @@
+import sampleImage from './sampleData/cat.png'
+
+
+//
```
Did you mean to leave a comment here?

alexander-huerta replied:
no, I will delete

RehReis commented on client/src/components/ProductDetails/StyleSelector.jsx:
```diff
@@ -0,0 +1,28 @@
+import css from './prodStyles.css';
+import StyleThumbnail from './StyleThumbnail.jsx'
+
+//props(styles) = array of objs. ea obj is a style
```
Did you mean to leave this comment in the final code? It might not be needed since you can understand it from the code.

RehReis replied:
(same for other similar comments around the PR)

alexander-huerta replied:
I just deleted the comments left in all files. I copied them into a google docs for my personal notes

RehReis commented on client/src/styles/styles.css:
```diff
@@ -0,0 +1,4 @@
+/* body {
```
Did you mean to leave this commented? This file seems unnecessary since it's not being used anywhere in the codebase.

alexander-huerta replied:
I left it incase we wanted to do any styling project wide. But im sure bootstrap will handle this. I will delete the file before re submitting a pull req

RehReis commented on webpack.config.js:
```diff
@@ -1,3 +1,5 @@
+// import css from './client/src/components/ProductDetails/prodStyles.css';
```
Did you mean to leave this commented?

alexander-huerta replied:
no. just deleted

RehReis reviewed (commented):
This looks like a great start. Could you fill in more information in the pull request description? Some screenshots of what this looks like might be helpful to understand it as well, since it's hard to visualize from just reading the code.

RehReis commented:
@alexander-huerta thanks for addressing the comments. Not sure if you saw the other note - could you also fill in the pull request description and add screenshots so we can get a sense of what this looks like?

alexander-huerta commented:
OVERVIEW OF CHANGES IN THIS PULL REQUEST:

In order for webpack to compile CSS and image formats, I downloaded the following dev dependencies: ‘css-loader’, ‘style-loader’, and ‘file loader’. I also added some configuration to the webpack config file to incorporate these loaders. 

In order to add some CSS styling to my components, I created  prodStyles.css file. I then imported this into each component that is taking advantage of css. For example in StyleThumbnail, the image is rendering in an oval shape. 

I also Created the various components my module will need. The structure is as follows:

ProductOverview
---->ImageGallery
-------->ImageThumbnails
---->StyleSelector
-------->StyleThumbnails

I also added some dummyData, and sample images. 


<img width="643" alt="productOverview" src="https://user-images.githubusercontent.com/59572783/147500502-24bb7343-e677-410d-9e36-1e0b61be91ca.png">

RehReis commented on client/src/components/ProductDetails/prodStyles.css:
```diff
@@ -0,0 +1,61 @@
+  height: auto;
+  border-radius: 50%;
+}
+/*
```
Did you mean to leave this commented?

RehReis commented on client/src/components/ProductDetails/ImageThumbnail.jsx:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import sampleImage from './sampleData/cat.png'
```
I see that ImageThumbnail is using its props as the source image instead of using sampleImage.
Did you mean to leave this line of code?

RehReis reviewed (commented):
All of this seems really good, @alexander-huerta!
A few suggestions I have:

I'd delete jpg files that aren't being used in the codebase. I couldn't leave a comment on the files, so I'm just listing them below. 
client/dist/client/src/components/ProductDetails/alex.jpg
client/dist/client/src/components/ProductDetails/fish-thumbnail.jpeg
client/dist/client/src/components/ProductDetails/sampleData/alex.jpg 

I'd suggest breaking Drop-Downs And Buttons section into a component(s) to make the ProductOverview component less complex.

alexander-huerta commented:
This pull request has been closed. A new request is being submitted with some improvements and refactors. Mainly the addition of hooks for state management
